### DISCUSSION TITLE
Add scope descriptions to plugin READMEs

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#1197] [Ideas]
Right now, a README of a plugin may have something alongside this in their "Supported specifications" section:

`🔑 (scopeless)` `read:org (optional)` `read:user (optional)` `read:packages (optional)` `repo (optional)`

While it shows what is and isn't required, does it not really tell you what each scope's purpose here would be.
What would `read:org` achieve for this plugin?
If I want to have repos of another org listed, do I only need a token with `read:org`, one with `repo` or both?

So, it could be a good idea to add a section within the README file to explain any listed scope and their purpose when used.
It would help reduce general confusion, especially for people who have no understanding of tokens, their scope and what is and isn't required for their needs.

**Comment by lowlighter** [comment]
It's partially documented here:
https://github.com/lowlighter/metrics/blob/c895cb042df5005143e3ed5091577706382249cc/.github/readme/partials/documentation/setup/action.md?plain=1#L25-L32

But it's true that it's still pretty vague. The optional scopes you listed are most of the time appended automatically by the docs generator
Things is that as far as I know, GitHub doesn't provide detailed access of each scope

Like I'm pretty confident that anything related to org requires `read:org` and anything with private data requires `repo` scopes.
For other it's a bit more blurry, like not sure if `read:packages` is actually used somewhere, `read:project` is required for new GitHub projects, but the legacy ones are requiring the `public_repo` scope, and for `read:user` I'm not sure what this covers (I remember it being used for 1 or 2 specific stats but not sure which ones actually)
